src/main/java/org/folio/linked/data/service/resource/impl/ResourceGraphServiceImpl.java

```diff
@@ -19,19 +18,27 @@
 import org.folio.linked.data.repo.ResourceRepository;
 import org.folio.linked.data.service.resource.ResourceGraphService;
 import org.folio.linked.data.util.JsonUtils;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Log4j2
 @Service
 @Transactional
-@RequiredArgsConstructor
 public class ResourceGraphServiceImpl implements ResourceGraphService {
 
   private final ResourceRepository resourceRepo;
   private final ResourceEdgeRepository edgeRepo;
   private final ResourceDtoMapper resourceDtoMapper;
 
+  public ResourceGraphServiceImpl(ResourceRepository resourceRepo,
+                                  ResourceEdgeRepository edgeRepo,
+                                  @Lazy ResourceDtoMapper resourceDtoMapper) {
```

Comment:
Since we started saving entities in our mappers we have circular dependency:
```
resourceGraphServiceImpl -> depends on resourceDtoMapperImpl
resourceDtoMapperImpl -> depends on singleResourceMapper
singleResourceMapperImpl -> depends on assigningSourceMapperUnit
assigningSourceMapperUnit -> depends on resourceMarcAuthorityServiceImpl
resourceMarcAuthorityServiceImpl -> depends on resourceGraphServiceImpl
```

```@Lazy``` annotation is on of the options to fix this issue (not working with ```@RequiredArgsConstructor```).

As an alternative option we can split ResourceGraphService to ResourceGraphReadService and ResourceGraphModifyService and move  ```resourceDtoMapper``` to ResourceGraphReadService.

@pkjacob @askhat-abishev @PBobylev let me know what do you guys think about it.

Reply:
> As an alternative option we can split ResourceGraphService to ResourceGraphReadService and ResourceGraphModifyService and move resourceDtoMapper to ResourceGraphReadService.

I like that option

@PBobylev  @askhat-abishev ?

Reply:
We had such issue before and we resolved it with Lazy:
org/folio/linked/data/mapper/dto/common/CoreMapperImpl.java:32

So I'm ok with it as it is now, considering that ResourceGraphService is not that big.

Reply:
@pkjacob i agree with Pavel, let me know what do you think.